### ISSUE TITLE
”TypeError: Cannot read property 'type' of undefined“

### ISSUE THREAD

**Issue by HamishMacEwan** [Xtrimmer/lndecode#3]
I believe the invoice string is correct, but the error ”TypeError: Cannot read property 'type' of undefined“ seems to cause lndecode to mark it incorrect.

`https://lndecode.com/?invoice=lnbc190n1p0q3tvapp58eg4t4wvauv7dw7hw8auhs3kvctej2amxyr789qtmuywgvh7a87qdp2d9hxc6twv5knzdeexcmnvdekx56njvesx5mnxwfc8yxqyz5vqcqzyssp5r09au7e74epce307xydezwdqxv6g5xh4m7xhr0slj42gr25d0ehqrzjqwgtt4zf9hp02vvw2ge6kt8t7m2gj9ygrge7765ud0xmkse6mxrdqzrytgqq9mqqqqqqqquyqqqqqksqrc9qy9qsqehre8cdlcg8ayydqd7vp63lt5rnj52d0rffpykr34fh7krwwczuxl8uteel048ps9ntzpryk6nzl23tfxlrtx2zsunuccdlkuklfc9sqspgzle`

**Comment by Xtrimmer** [comment]
I think what is happening here is that the decoder is unaware of some recent changes to the Bolt 11 specification. Specifically, this invoice contains the following two tagged fields that are not recognized because they have not been programmed into the decoder logic:

- s (16): data_length 52. This 256-bit secret prevents forwarding nodes from probing the payment recipient.
- 9 (5): data_length variable. One or more 5-bit values containing features supported or required for receiving this payment. See Feature Bits.

I need to update the code to recognize these two tagged fields and review any other recent additions that may need to be accounted for.

**Comment by Xtrimmer** [comment]
I've pushed a change set to include these new tags. It may take a day or two for the changes to propagate to all content delivery endpoints. I am able to hit the updated site where I am after clearing the browser cache. I think I interpreted the feature bits decoding rules correctly. Let me know if the result is what you are expecting from this transaction. If so then I will close the issue as fixed.
